src/main/java/de/dhbw/karlsruhe/students/mailflow/core/domain/email/Mailbox.java

```diff
@@ -41,10 +42,33 @@ public List<Email> getEmailList() {
     return emails.keySet().stream().toList();
   }
 
+  /**
+   * @deprecated, use {@code #markWithLabel} instead
+   *
+   * @param email
+   * @param isUnread
+   */
   public void deliverEmail(Email email, boolean isUnread) {
     this.emails.put(email, Set.of(isUnread ? Label.UNREAD : Label.READ));
   }
 
+  /**
+   * Deletes this e-mail from this mailbox. This action is idempotent.
+   *
+   * @param email
+   * @return {@link Optional#empty()} when the e-mail was not in this inbox, otherwise returns the
+   *         Set
```

Comment:
```suggestion
   * @return {@link Optional#empty()} when the e-mail was not in this inbox, otherwise returns the
   *         Set of labels the email had assigned
```